### PR TITLE
Apptainer rename

### PR DESCRIPTION
**Pull request by PaulCharlton**
Rename files/dirs and internal code as Apptainer from Singularity, preserving case

### CODE REVIEW

**Comment by cclerget** [comment]
looks like there are conflicts @PaulCharlton

File: dist/rpm/apptainer.spec.in
Comment (cclerget) [inline dist/rpm/apptainer.spec.in:137]: we need to keep `run-singularity` here otherwise it would break the feature of directly executing a SIF image

File: dist/debian/generate_manpages
Comment (cclerget) [inline dist/debian/generate_manpages:44]: `run-singularity` here too

File: dist/debian/rules
Comment (cclerget) [inline dist/debian/rules:74]: `run-singularity`

**Review by cclerget — COMMENTED** [review]
And `scripts/run-singularity` can't be renamed

File: dist/rpm/apptainer.spec.in
Comment (PaulCharlton) [reply dist/rpm/apptainer.spec.in:137]: is this an external dependency?   breaking change for existing scripts owned by users?

File: CHANGELOG.md
Comment (DrDaveD) [inline CHANGELOG.md]: No, we do not want to rename all the configuration variables.  We have in our announcement that we weren't going to change these.

**Review by DrDaveD — CHANGES_REQUESTED** [review]

**Comment by DrDaveD** [comment]
After you have this passing tests I would like to see the commits cleaned up so there aren't so many that needed correcting.  That also provides an opportunity to add the Sign-off.  I also think it would be better to break this up into smaller PRs.  For example start with just renaming files.

File: cmd/internal/cli/apptainer.go
Comment (DrDaveD) [inline cmd/internal/cli/apptainer.go:66]: No, we need to retain backward compatibility for environment variables.

**Review by DrDaveD — CHANGES_REQUESTED** [review]

**Comment by PaulCharlton** [comment]
noting ongoing governance discussion via the Apptainer/development slack channel

File: cmd/internal/cli/apptainer.go
Comment (PaulCharlton) [reply cmd/internal/cli/apptainer.go:66]: I invite us to defer this until resolution of issue #7 and make #7 a blocker for v1.0.0

File: CHANGELOG.md
Comment (PaulCharlton) [reply CHANGELOG.md]: I invite us to defer this into resolution of #7 and make #7 a blocker for v1.0.0

Comment (gmkurtzer) [reply CHANGELOG.md]: I concur with this idea. Thank you Paul!

**Comment by gmkurtzer** [comment]
I'm not familiar with DCO, what is the fix there?

**Comment by DrDaveD** [comment]
> I'm not familiar with DCO, what is the fix there?

That's the Developer Certificate of Origin check.  The fix is to have each commit have a "Signed-off-by" added with `git commit -s` or `git cherry-pick -s`.

**Comment by PaulCharlton** [comment]
I will squash and sign ...

File: dist/rpm/apptainer.spec.in
Comment (DrDaveD) [inline dist/rpm/apptainer.spec.in:136]: Need a %{_bindir}/singularity as well as %{_bindir}/apptainer.

File: cmd/internal/cli/apptainer.go
Comment (DrDaveD) [reply cmd/internal/cli/apptainer.go:66]: Yes so in the meanwhile I think no change should be made to environment variables.

File: e2e/env/regressions.go
Comment (DrDaveD) [inline e2e/env/regressions.go:135]: Change from ApptainerCE to Apptainer; the original must have been a mistake made during porting something from sylabs.

File: cmd/internal/cli/inspect.go
Comment (DrDaveD) [inline cmd/internal/cli/inspect.go:536]: The default definition file name should stay as Singularity.def for compatibility.

**Review by DrDaveD — CHANGES_REQUESTED** [review]
These are some preliminary things I have found.  There's a lot more to review yet.

File: cmd/internal/cli/inspect.go
Comment (PaulCharlton) [reply cmd/internal/cli/inspect.go:536]: of course you are correct, and these can be resolved by later Issue/PR rather than blocking this one

This is not a case where Athena needs to leap from Jupiter's head fully formed

File: e2e/env/regressions.go
Comment (PaulCharlton) [reply e2e/env/regressions.go:135]: of course you are correct, and these can be resolved by later Issue/PR rather than blocking this one

This is not a case where Athena needs to leap from Jupiter's head fully formed

File: dist/rpm/apptainer.spec.in
Comment (PaulCharlton) [reply dist/rpm/apptainer.spec.in:136]: of course you are correct, and these can be resolved by later Issue/PR rather than blocking this one

This is not a case where Athena needs to leap from Jupiter's head fully formed

**Review by ikaneshiro — COMMENTED** [review]
The urls that need fixing should be updated.

I think some of my comments around the build scripts are beyond renaming and require full refactoring. So a follow up PR is reasonable to me

File: cmd/internal/cli/pull.go
Comment (PaulCharlton) [reply cmd/internal/cli/pull.go:36]: submitting update

File: e2e/actions/actions.go
Comment (PaulCharlton) [reply e2e/actions/actions.go:395]: submitting update

File: e2e/actions/actions.go
Comment (PaulCharlton) [reply e2e/actions/actions.go:879]: submitting update

File: e2e/docker/docker.go
Comment (PaulCharlton) [reply e2e/docker/docker.go:404]: submitting update

File: internal/pkg/cache/cache.go
Comment (PaulCharlton) [reply internal/pkg/cache/cache.go:47]: submitting update

File: e2e/env/regressions.go
Comment (PaulCharlton) [reply e2e/env/regressions.go:135]: submitting update

File: internal/pkg/build/sources/base_environment.go
Comment (PaulCharlton) [reply internal/pkg/build/sources/base_environment.go:41]: this is resolved by #7

**Review by cclerget — CHANGES_REQUESTED** [review]

File: .github/dependabot.yml
Comment (PaulCharlton) [reply .github/dependabot.yml:13]: there is no merge conflict, so merging can proceed without a rebase

File: pkg/sylog/sylog_test.go
Comment (PaulCharlton) [reply pkg/sylog/sylog_test.go:292]: true -- however, please make this an issue ... amending the commit destroys the comment reference above

File: internal/pkg/runtime/engine/apptainer/process_linux.go
Comment (PaulCharlton) [reply internal/pkg/runtime/engine/apptainer/process_linux.go:894]: true -- however, please make this a generic issue to maintain backward compatibility containers ... amending the commit destroys the comment reference above.   Also, for the generic issue, we need to update the E2E tests to detect and enforce a backward compatibility rule.   Such E2E update does not belong in this PR

File: internal/pkg/build/sources/conveyorPacker_oci.go
Comment (PaulCharlton) [reply internal/pkg/build/sources/conveyorPacker_oci.go:369]: true -- however, please make this a generic issue to maintain backward compatibility containers ... amending the commit destroys the comment reference above.   Also, for the generic issue, we need to update the E2E tests to detect and enforce a backward compatibility rule.   Such E2E update does not belong in this PR

File: internal/pkg/build/sources/base_environment.go
Comment (PaulCharlton) [reply internal/pkg/build/sources/base_environment.go:332]: true -- however, please make this a generic issue to maintain backward compatibility containers ... amending the commit destroys the comment reference above.   Also, for the generic issue, we need to update the E2E tests to detect and enforce a backward compatibility rule.   Such E2E update does not belong in this PR

File: internal/pkg/build/sources/base_environment.go
Comment (PaulCharlton) [reply internal/pkg/build/sources/base_environment.go:41]: true -- however, please make this a generic issue to maintain backward compatibility containers ... amending the commit destroys the comment reference above.   Also, for the generic issue, we need to update the E2E tests to detect and enforce a backward compatibility rule.   Such E2E update does not belong in this PR

File: internal/pkg/build/metadata.go
Comment (PaulCharlton) [reply internal/pkg/build/metadata.go:178]: perhaps true -- however, this is a question of "default app" which the end user can already override during container build process, yes?  -- however, please make this a generic issue to maintain backward compatibility containers ... amending the commit destroys the comment reference above.   Also, for the generic issue, we need to update the E2E tests to detect and enforce a backward compatibility rule.   Such E2E update does not belong in this PR

File: cmd/internal/cli/inspect.go
Comment (PaulCharlton) [reply cmd/internal/cli/inspect.go:536]: true -- however, please make this a generic issue to maintain backward compatibility containers ... amending the commit destroys the comment reference above.   Also, for the generic issue, we need to update the E2E tests to detect and enforce a backward compatibility rule.   Such E2E update does not belong in this PR

File: dist/rpm/apptainer.spec.in
Comment (PaulCharlton) [reply dist/rpm/apptainer.spec.in:136]: true -- however, please make this a generic issue to maintain backward compatibility containers ... amending the commit destroys the comment reference above.   Also, for the generic issue, we need to update the E2E tests to detect and enforce a backward compatibility rule.   Such E2E update does not belong in this PR

File: .github/dependabot.yml
Comment (cclerget) [reply .github/dependabot.yml:13]: the point is the changes here are going to reintroduce lines which have been removed previously by #30 , see https://github.com/apptainer/apptainer/blob/main/.github/dependabot.yml for the updated content

File: pkg/sylog/sylog_test.go
Comment (cclerget) [reply pkg/sylog/sylog_test.go:292]: Could you just apply the fix, I'll make it as a suggestion but feel free to just replicate it on your branch, I don't think it justify a follow-up issue

File: cmd/internal/cli/apptainer.go
Comment (PaulCharlton) [reply cmd/internal/cli/apptainer.go:66]: deferred to issue #7

File: dist/rpm/apptainer.spec.in
Comment (PaulCharlton) [reply dist/rpm/apptainer.spec.in:136]: deferred to issue #39

File: cmd/internal/cli/inspect.go
Comment (PaulCharlton) [reply cmd/internal/cli/inspect.go:536]: deferred to #40

File: internal/pkg/build/metadata.go
Comment (PaulCharlton) [reply internal/pkg/build/metadata.go:178]: resolved by #40

File: internal/pkg/build/sources/base_environment.go
Comment (PaulCharlton) [reply internal/pkg/build/sources/base_environment.go:332]: deferred to #40

File: internal/pkg/runtime/engine/apptainer/process_linux.go
Comment (PaulCharlton) [reply internal/pkg/runtime/engine/apptainer/process_linux.go:894]: deferred to #40

File: internal/pkg/build/sources/base_environment.go
Comment (PaulCharlton) [reply internal/pkg/build/sources/base_environment.go:41]: deferred to #7

File: pkg/sylog/sylog_test.go
Comment (PaulCharlton) [reply pkg/sylog/sylog_test.go:292]: deferred to #42 [will inspect and look for other instances of this coding practice]

File: examples/legacy/2.1/arch.def
Comment (PaulCharlton) [reply examples/legacy/2.1/arch.def:2]: deferred to #44

**Review by ikaneshiro — APPROVED** [review]

File: cmd/internal/cli/actions.go
Comment (DrDaveD) [inline cmd/internal/cli/actions.go:51]: Should be 1.0+

Comment (PaulCharlton) [reply cmd/internal/cli/actions.go:51]: @dwd please write an issue for deferral and mark this resolved

**Review by cclerget — APPROVED** [review]

**Review by DrDaveD — APPROVED** [review]
Although there are many problems remaining with this, the group has decided to defer all those to changes to other issues required by the 1.0.0 milestone.

File: cmd/internal/cli/startvm.go
Comment (DrDaveD) [reply cmd/internal/cli/startvm.go:75]: Defer to #47

File: docs/content.go
Comment (DrDaveD) [reply docs/content.go:455]: Defer to #47

File: e2e/docker/docker.go
Comment (DrDaveD) [reply e2e/docker/docker.go:423]: Defer to #47

File: internal/pkg/cache/cache.go
Comment (DrDaveD) [reply internal/pkg/cache/cache.go:185]: Defer to #47

File: internal/pkg/util/fs/files/action_scripts.go
Comment (DrDaveD) [reply internal/pkg/util/fs/files/action_scripts.go:111]: Defer to #47

File: examples/plugins/cli-plugin/README.md
Comment (DrDaveD) [reply examples/plugins/cli-plugin/README.md:13]: Defer to #47

File: cmd/internal/cli/actions.go
Comment (DrDaveD) [reply cmd/internal/cli/actions.go:51]: Defer to #47

**Pull request merged**
